src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SamlSecurityToken.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using CoreWCF.Security;
+using MSIdentitySAML = Microsoft.IdentityModel.Tokens.Saml;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    public class SamlSecurityToken : SecurityToken
+    {
+        protected SamlSecurityToken()
+        {
+        }
+
+        internal SamlSecurityToken(MSIdentitySAML.SamlSecurityToken samlSecurityToken)
+        {
+            InternalSamlSecurityToken = samlSecurityToken;
+            Assertion = InternalSamlSecurityToken.Assertion;
+        }
+
+        public SamlSecurityToken(MSIdentitySAML.SamlAssertion assertion)
+        {
+            Initialize(assertion);
+        }
+
+        protected void Initialize(MSIdentitySAML.SamlAssertion assertion)
+        {
+            if (assertion != null)
+            {
+                Assertion = assertion;
+                // this.assertion.MakeReadOnly();
+                SamlStatements = new List<InternalSamlSubjectStatement>();
+            }
+            else
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(assertion));
+        }
```

Comment:
Add braces as it looks wrong to have braces for the if block but not the else block.